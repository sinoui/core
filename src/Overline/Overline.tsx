import { TextContent, typographyPub } from '../utils/typographyPub';

const Overline = typographyPub('p', 'overline')(TextContent);
export default Overline;
