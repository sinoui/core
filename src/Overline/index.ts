import createTypography from '../Typography/createTypography';

export default createTypography('overline', 'Overline');
