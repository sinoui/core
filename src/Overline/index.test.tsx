import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import Overline from './index';

/**
 * Overline组件单元测试
 */
describe('文字排版 单元测试', () => {
  afterEach(cleanup);

  it('指定不同的根元素', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Overline as="div">这是一个文字。</Overline>
      </ThemeProvider>,
    );

    const text = container.querySelector('div');
    expect(text).toHaveTextContent('这是一个文字。');
  });

  it('ref 指向根元素', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Overline data-testid="typo" ref={ref}>
          这是一个文字。
        </Overline>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toEqual(ref.current);
  });
  it('设置align文本对齐方式', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Overline data-testid="typo" align="center">
          这是一个文字。
        </Overline>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toHaveStyle(`text-align:center`);
  });
});

/**
 * Overline组件快照测试
 */
describe('文字排版 快照测试', () => {
  it('基本文字排版', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Overline>Overline</Overline>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置属性', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Overline as="h2">Overline</Overline>
          <Overline align="center">Overline</Overline>
          <Overline noWrap>Overline</Overline>
          <Overline color="success">Overline</Overline>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('颜色设置', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Overline>Overline</Overline>
          <Overline color="primary">Overline</Overline>
          <Overline color="secondary">Overline</Overline>
          <Overline color="textPrimary">Overline</Overline>
          <Overline color="textSecondary">Overline</Overline>
          <Overline color="error">Overline</Overline>
          <Overline color="warning">Overline</Overline>
          <Overline color="success">Overline</Overline>
          <Overline color="info">Overline</Overline>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
