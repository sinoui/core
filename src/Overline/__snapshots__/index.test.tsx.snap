// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文字排版 快照测试 基本文字排版 1`] = `
<p
  className="Typography__TypographyWrapper-sc-12agh80-0 ekHPVY sinoui-typography sinoui-typography--overline"
  color="textPrimary"
  type="overline"
>
  Overline
</p>
`;

exports[`文字排版 快照测试 设置属性 1`] = `
Array [
  <h2
    className="Typography__TypographyWrapper-sc-12agh80-0 ekHPVY sinoui-typography sinoui-typography--overline"
    color="textPrimary"
    type="overline"
  >
    Overline
  </h2>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 eApNRY sinoui-typography sinoui-typography--overline"
    color="textPrimary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 loutZp sinoui-typography sinoui-typography--overline"
    color="textPrimary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 jLIVtw sinoui-typography sinoui-typography--overline"
    color="success"
    type="overline"
  >
    Overline
  </p>,
]
`;

exports[`文字排版 快照测试 颜色设置 1`] = `
Array [
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 ekHPVY sinoui-typography sinoui-typography--overline"
    color="textPrimary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 bqTgsP sinoui-typography sinoui-typography--overline"
    color="primary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 dqvyw sinoui-typography sinoui-typography--overline"
    color="secondary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 ekHPVY sinoui-typography sinoui-typography--overline"
    color="textPrimary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 bvmFLd sinoui-typography sinoui-typography--overline"
    color="textSecondary"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 dYpqqh sinoui-typography sinoui-typography--overline"
    color="error"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 huDQKM sinoui-typography sinoui-typography--overline"
    color="warning"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 jLIVtw sinoui-typography sinoui-typography--overline"
    color="success"
    type="overline"
  >
    Overline
  </p>,
  <p
    className="Typography__TypographyWrapper-sc-12agh80-0 hSzcQM sinoui-typography sinoui-typography--overline"
    color="info"
    type="overline"
  >
    Overline
  </p>,
]
`;
